pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -57,7 +58,7 @@ import (
 
 var (
 	edsClusterMutex sync.RWMutex
-	edsClusters     = map[string]*EdsCluster{}
+	edsClusters     = map[string]map[model.Locality]*EdsCluster{}
```

Comment:
I thought about this  - and still fail  to understand why is it needed.

The purpose of edsClusters is to know which Envoys need to be pushed config data on incremental update. The incremental update knows the hostname ( doesn't deal with locality or subsets - just the service ) - and updates any envoy that is watching clusters belonging to the service.

With the network scopes and other changes - we'll need to recompute the result list anyways for each envoy.

So why would we add a locality in this map ????

Reply:
If this is about caching LocalityEndpointAssignemnt per zone - it still doesn't make sense, this is very dynamic and the cache is unlikely to help, and will need to go away anyways for the other use cases.

If you do need to use the cache - you can use still use the list and select based on locality ( will need to use the envoy proto to access the data from the cached object ).